[Joanna] iP

cfyjoanna opened the pull request:
# Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Let Duke help you to store your plans so you never have to. Duke is
- text-based
- intuitive for everyone
- *completely **free***!

All you have to do is:
1. download it from [here](https://github.com/cfyjoanna/ip)
2. open the app
3. add your todos, deadlines and events
4. let Duke take care of the rest! :hugs:

## Features
- [x] manage tasks
- [ ] set reminders (coming soon)

## For programmers
You can change Duke to behave however you like too. Here's the `main` method:

```java
public class Duke {
    // ...
    public static void main(String[] args) {
        new Duke().run();
    }
}
```

Commit:
Add Gradle support

cfyjoanna committed:
Level-1 increment

cfyjoanna committed:
Revert "Level-1 increment"

This reverts commit 7506b730e4e13d1d7be5217db119bf7b5fe05a1a.

cfyjoanna committed:
Level-1 increment

cfyjoanna committed:
Level 2 increment

cfyjoanna committed:
Level 3 increment

cfyjoanna committed:
Level 4 increment

cfyjoanna committed:
added testing

cfyjoanna committed:
Level 5 increment

cfyjoanna committed:
Level 6 increment

cfyjoanna committed:
branch Level 7

cfyjoanna committed:
add date formatter

cfyjoanna committed:
Merge branch 'branch-Level-7'

cfyjoanna committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

cfyjoanna committed:
add TaskList class

cfyjoanna committed:
add more OOP

cfyjoanna committed:
create duke package

cfyjoanna committed:
Merge branch 'add-gradle-support'

cfyjoanna committed:
add JUnit tests

cfyjoanna committed:
add JavaDocs

cfyjoanna committed:
follow coding standard

cfyjoanna committed:
add find

cfyjoanna committed:
Merge branch 'branch-A-JavaDoc'

cfyjoanna committed:
Merge branch 'branch-A-CodingStandard'

cfyjoanna committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Task.java
#	src/main/java/duke/TaskList.java

cfyjoanna committed:
add incomplete GUI

cfyjoanna committed:
add responsiveness to GUI

cfyjoanna committed:
fix text cut off

cfyjoanna committed:
add welcome message

cfyjoanna committed:
Merge branch 'branch-Level-10'

cfyjoanna committed:
add checkstyle

cfyjoanna committed:
remove unnecessary code

TZL0 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,42 @@
+    /**
+     * Prints out goodbye message.
+     */
+    public String showGoodbye() {
```
Is it better to name it getGoodbyeMessage?

cfyjoanna replied:
Nice suggestion! Will make this and the other changes.

TZL0 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,42 @@
+    /**
+     * Prints out welcome message.
+     */
+    public String showWelcome() {
```
Is it better to name it getWelcomeMessage?

TZL0 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,126 @@
+     * @param input The user's input.
+     * @param type The type of task.
+     */
+    public String processTask(String input, String type) {
```
Is it better to name it parseTask()?

TZL0 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,117 @@
+     * @param input The String to search for.
+     * @return String containing a numbered list of the results.
+     */
+    public String find(String input) {
```
Does it really returns a List of Tasks?

cfyjoanna replied:
The intended meaning was a text containing all the results found in String format (in numbered list form), and not the List object. Is there another way I should phrase this to make it clearer? Thanks!

cfyjoanna committed:
rename methods

cfyjoanna committed:
rename methods

cfyjoanna committed:
enable assertions

cfyjoanna committed:
add default in switch statements

cfyjoanna committed:
add assertions

RichDom2185 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,110 @@
+            reader.close();
+        } catch (FileNotFoundException e) {
+            // Create the empty file if there is no existing file.
+            File file = new File(filePath);
+            try {
+                file.getParentFile().mkdirs();
+                file.createNewFile();
```
Do you think it would be better if we declare `file` in the same scope as the `try` block?

```suggestion
            try {
                File file = new File(filePath);
                file.getParentFile().mkdirs();
                file.createNewFile();
```

cfyjoanna replied:
Thanks for pointing these out! Will rectify.

RichDom2185 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,110 @@
+            for (int i = 0; i < tasks.size(); i++) {
+                String str = tasks.get(i).toString();
+                StringBuilder task = new StringBuilder();
+                task.append(str.charAt(1) + " // ");
+                if (str.charAt(4) == 'X') {
```
Maybe constants could be used here instead of so-called "magic numbers" for clarity?

RichDom2185 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,54 @@
+     * @return Boolean if the input is found.
+     */
+    public boolean contains(String input) {
+        return this.description.toLowerCase().contains(input.toLowerCase());
```
Would it be better if we avoid the unnecessary use of the `this` keyword, since `description` is not shadowed?

RichDom2185 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,54 @@
+     * Marks task as done.
+     */
+    public void markAsDone() {
+        this.isDone = true;
```
Ditto on `this` and shadowing.

RichDom2185 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,54 @@
+     * Marks task as undone.
+     */
+    public void markAsUndone() {
+        this.isDone = false;
```
Ditto on `this` and shadowing.

RichDom2185 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,54 @@
+     */
+    @Override
+    public String toString() {
+        return "[" + this.getStatusIcon() + "] " + this.description;
```
Ditto on `this` and shadowing. The issue is also present in a couple other places in the code.

RichDom2185 reviewed (commented):
Overall, looks really good. Great attention to the coding standard!

cfyjoanna committed:
Updated code to reflect Code Quality

Removed unnecessary uses of the this keyword.

Avoided use of magic literals. Named constants were used in place
of them.

To further improve code quality, long methods could be extracted
into multiple smaller methods.

cfyjoanna committed:
Merge branch 'master' into branch-A-Assertions

cfyjoanna committed:
Merge pull request #1 from cfyjoanna/branch-A-Assertions

Branch a assertions

cfyjoanna committed:
Merge branch 'master' of https://github.com/cfyjoanna/ip

cfyjoanna committed:
Merge pull request #2 from cfyjoanna/branch-A-CodeQuality

Updated code to reflect Code Quality

cfyjoanna committed:
Merge branch 'master' of https://github.com/cfyjoanna/ip

cfyjoanna committed:
add help command

cfyjoanna committed:
modify welcome message

cfyjoanna committed:
Update code quality

cfyjoanna committed:
add Ui.png

cfyjoanna committed:
add user guide

cfyjoanna committed:
rename mainClassName to duke.Launcher

cfyjoanna committed:
Removed lines to fix bugs
